cmd/nvidia-dra-controller/imex.go

```diff
@@ -79,21 +87,25 @@ func StartIMEXManager(ctx context.Context, config *Config) (*ImexManager, error)
 		UID:        pod.UID,
 	}
 
-	// Create the manager itself
-	m := &ImexManager{
-		clientset: clientset,
+	// Create a new set of DriverResources
+	driverResources := &resourceslice.DriverResources{
+		Pools: make(map[string]resourceslice.Pool),
```

Comment:
did you run into an error here? We removed this approach on the past PR. I wonder if you are running into issues by not doing this?

Reply:
We should have needed it ever since we moved away from the custom `DeepCopy()` function. I missed it in the later rounds of reviews. I'm surprised it wasn't crashing for you in your testing without it.